Calcul des vagues de chaleur.

edemartel opened the pull request:
Importation des données météo d'Environnement Canada pour chacune des stations météorologiques du Québec depuis le 1er janvier 1990.

On considère une vague de chaleur comme une période d'au moins trois jours consécutifs où la température maximale atteint au moins 30 degrés.

Pour chaque vague de chaleur ainsi déterminée, on créé une catastrophe de type "vague de chaleur" ayant une position (approximative) située sur la station, une date égalant la date de début de la séquence de jours chauds, et une sévérité selon la durée en jours.

Pour le moment, j'utilise les icônes de feu de forêt comme icône pour les vagues de chaleur, mais ça sera à modifier.

Puisque les données météorologiques totalisent plusieurs centaines de MB, je ne versionne que le CSV résultant du calcul de vagues de chaleur. Pour obtenir vous même les données, voir le README dans tools/data/raw_climate_data.

edemartel committed:
Calcul des vagues de chaleur.

Importation des données météo d'Environnement Canada pour chacune des
stations météorologiques du Québec depuis le 1er janvier 1990.

On considère une vague de chaleur comme une période d'au moins trois
jours consécutifs où la température maximale atteint au moins 30 degrés.

Pour chaque vague de chaleur ainsi déterminée, on créé une catastrophe
de type "vague de chaleur" ayant une position (approximative) située
sur la station, une date égalant la date de début de la séquence de
jours chauds, et une sévérité selon la durée en jours.

Pour le moment, j'utilise les icônes de feu de forêt comme icône pour
les vagues de chaleur, mais ça sera à modifier.

Puisque les données météorologiques totalisent plusieurs centaines
de MB, je ne versionne que le CSV résultant du calcul de vagues de
chaleur. Pour obtenir vous même les données, voir le README dans
tools/data/raw_climate_data.

Drahakar commented on tools/fetch_climate_data.py:
```diff
@@ -0,0 +1,110 @@
+            'end': int(dlyr)
+        }
+
+HEADERS = {
```
Pas utilisé

Drahakar commented on tools/fetch_climate_data.py:
```diff
@@ -0,0 +1,110 @@
+        station = stations[station_id]
+        args = [sys.executable, script_path, station_id, str(
+            station['start']), str(station['end']), temp_path]
+        process = subprocess.Popen(args)
```
Si je comprends bien, le but ici c'est de lancer plusieurs appels en parallel; pourquoi ne pas utiliser multiprocessing?

https://docs.python.org/3/library/multiprocessing.html

edemartel replied:
Faque en implémentant ça comme ça, ça m'a permis de réaliser que j'avais un bogue dans mon code qui faisait en sorte que certaines vagues de chaleur n'étaient pas identifiées comme telles.

edemartel committed:
Multiprocessing c'est mieux.

edemartel committed:
README pour les données brutes.

edemartel committed:
Récupération incrémentale des données d'une station.

Également, correction d'un bogue qui faisait en sorte que le premier jour
d'un mois était sauté, menant à des vagues de chaleur incorrectement
identifiées à plusieurs reprises.

edemartel committed:
Ajustement de formats.

edemartel committed:
Plus de filtrage par sévérité sur les vagues de chaleur.

Si on trouve qu'on en a trop plus tard on avisera.
